######################################################################
#
# chr.table.R
#
# 04-feb-2015 Debora
#
# To run in R from /raid/genevol/1kg/phase3/data/chrN
# 
# Creates a dataframe for chromosome N containing coding SNPs in lines
# and the following columns:
#
# CHR POS ID REF ALT refgene_GENE refgene_ANNOTATION encode_GENE
# encode_ANNOTATION ANC
# AF_ALT_all(*3) AF_ALT_POP(*26*3) AF_ALT1_noadm(*3)
# FST_overall FST_num_overall FST_den_overall
# FST_noadmix FST_num_noadmix FST_den_noadmix
# FST_POP1-POP2(*325) FST_num_POP1-POP2(*325) FST_den_POP1-POP2(*325)
#
######################################################################

library(stringr)
library(reshape2)

# get chromosome number
this.chr <- str_match(getwd(), "chr(.+?)$")
print(paste("Starting ",this.chr[,1]))

#-------------------------------------------------------------------------------
# GENERAL INFO ABOUT SNPS

# read first columns of vcf file
system.time(chr.table <- read.table(paste0("../phase3_chr/ALL.", this.chr[,1],
                ".phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz"),
                        colClasses = c(rep("integer", 2), "character",
                                       rep("factor", 2), "NULL", "NULL",
                                       "character", rep("NULL", 2505))
                        ))
print("Finished reading VCF")
# Time to read chr22 vcf:
#    user  system elapsed
# 900.380   0.820 905.928

colnames(chr.table) <- c("CHR", "POS", "ID", "REF", "ALT", "INFO")
# separate INFO field from vcf file
info <- chr.table[, 6]
chr.table <- chr.table[,-6]

# get variant type ( SNP, INDEL or SV) from vcf INFO field
VT <- str_match(info, "VT=(.+?)(?:;|$)")[, 2]

print("Reading annotation files")
# read annotation files generated by annovar
# and merged by merge_annovar_output.pl
for(database in dir("../annovar_output/")){
    if(database=="refGene"){
        dbshort <- "refgene"
    }
    else if(database=="wgEncodeGencodeBasicV19"){
        dbshort <- "encode"
    }
    else{
        dbshort <- database
    }
    annotation <- read.table(paste0("../annovar_output/",database,"/mergeanno_",
                                    this.chr[, 1], ".bz2"),
                             col.names = c("position", paste0(dbshort, "_GENE"),
                                           paste0(dbshort,"_ANNOTATION")),
                             colClasses = c("integer", "character", "character"))
    # add annotation to chr.table
    chr.table <- merge(x = chr.table, y = annotation,
                       by.x = "POS", by.y = "position", all.x = TRUE)
}

codingSNPindex <- which(VT=="SNP" & 
                        (!grepl("intergenic", chr.table$refgene_ANNOTATION) |
                        !grepl("intergenic", chr.table$encode_ANNOTATION)))

chr.table <- chr.table[codingSNPindex,]

# Ancestral allele (useful to calcultate DAF (derived allele frequency))
chr.table$ANC <- str_match(info[codingSNPindex], "(?:^|;)AA=(.*?)\\|")[, 2]

#-------------------------------------------------------------------------------
# ALLELE FREQUENCIES

# include column for frequency of the reference allele overall pops from vcf file
print("Reading allele frequency files")

AF_all <- str_match(info[codingSNPindex], "(?:^|;)AF=(.+?)(?:;|$)")[, 2]
## the above object will be a vector of strings containing the frequency of all
## alternative alleles in coding SNPs.
## Some of them (triallelic sites) will have two numbers separated by a comma
## (or 3 in case of 4-allelic sites).
## Structural variants (SV) may have more than
## 3 alternative alleles, so I'm filtering by coding SNPs only.

## split each element of vector above by ','.
chr.table[c('AF_ALT1_all', 'AF_ALT2_all', 'AF_ALT3_all')] <-
    colsplit(AF_all, ',', c('af1','af2','af3'))


# read files with frequency per population calculated using vcftools
files_frq <- list.files(pattern = ".frq")

noadmix_count <- data.frame(ALT1 = rep(0, length(codingSNPindex)),
                            ALT2 = rep(0, length(codingSNPindex)),
                            ALT3 = rep(0, length(codingSNPindex)),
                            TOTAL = 0)

for (file in files_frq){
    cat(file," ")
    temp <- read.table(file, row.names = NULL,
                       colClasses=c(rep("numeric", 4), rep("character", 4), rep("NULL",30)),
                       fill=TRUE)[codingSNPindex,]
    pop <-  str_match(file, "^[A-Z]{3}")
    af <- as.numeric(str_match(temp[, 6], ":(.+)")[, 2])/temp[, 4]
    chr.table[, paste0("AF_ALT1_", pop)] <- af
    af <- as.numeric(str_match(temp[, 7], ":(.+)")[, 2])/temp[, 4]
    chr.table[, paste0("AF_ALT2_", pop)] <- af
    af <- as.numeric(str_match(temp[, 8], ":(.+)")[, 2])/temp[, 4]
    chr.table[, paste0("AF_ALT3_", pop)] <- af
    # Calculate global frequency excluding admixed pops
    if (!pop %in% c("ACB","ASW","MXL","CLM","PUR","PEL")){
        noadmix_count$ALT1 <- noadmix_count$ALT1 + as.numeric(str_match(temp[,6], ":(.+)")[,2])
        noadmix_count$ALT2 <- noadmix_count$ALT2 + as.numeric(str_match(temp[,7], ":(.+)")[,2])
        noadmix_count$ALT3 <- noadmix_count$ALT3 + as.numeric(str_match(temp[,8], ":(.+)")[,2])
        noadmix_count$TOTAL <- noadmix_count$TOTAL + as.numeric(temp[,4])
  }
}

chr.table$AF_ALT1_noadm <- noadmix_count$ALT1/noadmix_count$TOTAL
chr.table$AF_ALT2_noadm <- noadmix_count$ALT2/noadmix_count$TOTAL
chr.table$AF_ALT3_noadm <- noadmix_count$ALT3/noadmix_count$TOTAL

#-------------------------------------------------------------------------------

print("Saving chr.table")
chr.table[chr.table == ""] <- NA
save("chr.table", file = paste0(this.chr[, 1], ".table.RData"))
save("codingSNPindex", file = paste0(this.chr[,1], ".codingSNPs.RData"))
